### PR TITLE
shared/killall: exclude pid1 helper processes from killing spree

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/shared/killall.c [hunk 1]
```diff
@@ -97,21 +96,21 @@ static bool ignore_proc(const PidRef *pid, bool warn_rootfs) {
         if (uid != 0)
                 return false;
 
+        (void) pidref_get_comm(pid, &comm);
```
Comment (bluca) [inline src/shared/killall.c:99]: isn't this under the control of the process being killed?

Comment (YHNdnzj) [reply src/shared/killall.c:99]: But it's limited to root? Also this isn't security sensitive I'd say, since for some errors we just return true.

Comment (bluca) [reply src/shared/killall.c:99]: Can the existing mechanism with the `@` be used instead, so that the same thing is used everywhere?

Comment (YHNdnzj) [reply src/shared/killall.c:99]: `sd-clone` uses `clone()` directly, and to pass this bit of information is kinda hard. (The explanation is included in the commit msg)

**Pull request closed without merging**